app/league/all/page.test.tsx

```diff
@@ -0,0 +1,41 @@
+import { render, screen, waitFor } from '@testing-library/react';
+import Leagues from './page';
+
+jest.mock('@/store/dataStore', () => ({
+  useDataStore: jest.fn(),
+}));
+
+jest.mock('@/utils/utils', () => ({
+  getUserLeagues: jest.fn(),
+}));
+
+jest.mock('@/api/apiFunctions', () => ({
+  getGameWeek: jest.fn(),
+}));
+
+describe('Leagues Component', () => {
+  const mockUser = { id: '123', leagues: [] };
+  const mockUseDataStore = require('@/store/dataStore').useDataStore;
+  const mockGetUserLeagues = require('@/utils/utils').getUserLeagues;
+  const mockGetGameWeek = require('@/api/apiFunctions').getGameWeek;
+
+  beforeEach(() => {
+    mockUseDataStore.mockReturnValue({ user: mockUser });
```

Comment:
Instead of returning the data here, the mock on line 5 should return this as the default. Then, if you need to override the return, the individual test can do that.